src/node_options.cc

```diff
@@ -0,0 +1,335 @@
+      {
+        if (!is_eval)
+          is_eval = (rc == 'e');
+
+        if (!is_print)
+          is_print = (rc == 'p');
+        const char* name = is_eval ? "eval" : "print";
+        NodeOptions::print_eval = NodeOptions::print_eval || is_print;
+        if (is_eval == true) {
+          NodeOptions::eval_string = argv[index + 1];
+          args_consumed += 1;
+          if (NodeOptions::eval_string == nullptr) {
+            fprintf(stderr, "%s: %s requires an argument\n", argv[0], name);
+            exit(9);
+          }
+        } else {
+          if ((index + 1 < nargs) &&
+              argv[index + 1] != nullptr &&
+              argv[index + 1][0] != '-') {
+            NodeOptions::eval_string = argv[index + 1];
+            args_consumed += 1;
+            if (strncmp(NodeOptions::eval_string, "\\-", 2) == 0) {
+              // Starts with "\\-": escaped expression, drop the backslash.
+              NodeOptions::eval_string += 1;
+            }
+          } else if ((index + 2 < nargs) &&
+                      argv[index + 2] != nullptr &&
+                      argv[index + 2][0] != '-') {
+            NodeOptions::eval_string = argv[index + 2];
+            args_consumed += 2;
+            if (strncmp(NodeOptions::eval_string, "\\-", 2) == 0) {
+              // Starts with "\\-": escaped expression, drop the backslash.
+              NodeOptions::eval_string += 1;
+            }
+          }
+        }
+        break;
+      }
+      case 'i':
+        NodeOptions::force_repl = true;
+        break;
+      case 'r':
+      {
+        const char* module = argv[index + 1];
+        args_consumed += 1;
+        if (module == nullptr) {
+          fprintf(stderr, "%s: %s requires an argument\n", argv[0], arg);
+          exit(9);
+        }
+        local_preload_modules[NodeOptions::preload_module_count++] = module;
+        break;
+      }
+      case OPT_NO_DEPRECATION:
+        NodeOptions::no_deprecation = true;
+        break;
+      case OPT_THROW_DEPRECATION:
+        NodeOptions::throw_deprecation = true;
+        break;
+      case OPT_TRACE_DEPRECATION:
+        NodeOptions::trace_deprecation = true;
+        break;
+      case OPT_TRACE_SYNC_IO:
+        NodeOptions::trace_sync_io = true;
+        break;
+      case OPT_V8_OPTIONS:
+        new_v8_argv[new_v8_argc] = "--help";
+        new_v8_argc += 1;
+        break;
+      case OPT_ABORT_UNCAUGHT:
+        NodeOptions::abort_on_uncaught_exception = true;
+        break;
+      case OPT_EXPOSE_INTERNALS:
+        // pass through
+        break;
+      case OPT_DEBUG:
+      {
+        NodeOptions::use_debug_agent = true;
+        if (optarg != nullptr) {
+          port = const_cast<const char*>(optarg);
+        }
+        break;
+      }
+      case OPT_DEBUG_BRK:
+      {
+        NodeOptions::use_debug_agent = true;
+        NodeOptions::debug_wait_connect = true;
+        if (optarg != nullptr) {
+          port = const_cast<const char*>(optarg);
```

Comment:
This is not very sound.  The string that `optarg` points to may get clobbered by later updates.  It would be best to process it immediately.

(Also, you don't need the const_cast.)